unity_jobs_simulator/Assets/Scripts/Robot_Controller.cs

```diff
@@ -0,0 +1,273 @@
+            }
+        }
+        Debug.Log("We just added " + _Joints.Count + " ArticulationBodies to the joints array");
+
+        if(stiffness.Count != _Joints.Count || damping.Count != _Joints.Count) // Error message if we don't have correct number of inputs
+        {
+            Debug.Log("Expected " + _Joints.Count + " inputs and Got " + stiffness.Count + " and " + damping.Count + " for the stiffness and damping lists.");
+        }
+
+        if(setAllJointDriveValues) // We set the Stiffness and Damping values to override the preset if told to do so
+        {
+            for(int i=0; i < _Joints.Count ; i++) // Runs through all the joints to set each value to the xDrive for each joint
+            {
+                ArticulationDrive drive = _Joints[i].xDrive;
+                drive.stiffness = stiffness[i];
+                drive.damping = damping[i];
+                _Joints[i].xDrive = drive;
+            }
+        }
+    }
+
+    void Update() //This only updates the target position variable depending on the input we receive for that frame
+    {
+        if(Input.GetKeyDown(KeyCode.R) || Input.GetKeyDown(KeyCode.Alpha0) || Input.GetKeyDown(KeyCode.Keypad0))
+        {
+            targetPos = new List<float>() {0, 0, 0, 0, 0, 0};
+        }
+        if(Input.GetKeyDown(KeyCode.A))
+        {
+            targetPos = new List<float>(){100, -60, 120, 180, -90, 45};
+        }
+        if(Input.GetKeyDown(KeyCode.S))
+        {
+            targetPos = new List<float>(){-45, -30, 15, 0, 90, 90};
+        }
+        if(Input.GetKeyDown(KeyCode.Keypad1) || Input.GetKeyDown(KeyCode.Alpha1))
+        {
+            targetPos = new List<float>() {90, 0, 0, 0, 0, 0};
+        }
+        if(Input.GetKeyDown(KeyCode.Keypad2) || Input.GetKeyDown(KeyCode.Alpha2))
+        {
+            targetPos = new List<float>() {0, -90, 0, 0, 0, 0};
+        }
+        if(Input.GetKeyDown(KeyCode.Keypad3) || Input.GetKeyDown(KeyCode.Alpha3))
+        {
+            targetPos = new List<float>() {0, 0, -90, 0, 0, 0};
+        }
+        if(Input.GetKeyDown(KeyCode.Keypad4) || Input.GetKeyDown(KeyCode.Alpha4))
+        {
+            targetPos = new List<float>() {0, 0, 0, 90, 0, 0};
+        }
+        if(Input.GetKeyDown(KeyCode.Keypad5) || Input.GetKeyDown(KeyCode.Alpha5))
+        {
+            targetPos = new List<float>() {0, 0, 0, 0, -90, 0};
+        }
+        if(Input.GetKeyDown(KeyCode.Keypad6) || Input.GetKeyDown(KeyCode.Alpha6))
+        {
+            targetPos = new List<float>() {0, 0, 0, 0, 0, 90};
+        }
+    }
+
+    void FixedUpdate()
+    {
+        updateCurrentDrivePos(); // We need to update the current position of the Robot to ensure we know exactly where it is
+        
+        if(samePos(_currentDrivePos, targetPos)) // If we are at the preset target, FixedUpdate shouldn't run so we return nothing
```

Comment:
You should be checking the currentPos array for the actual joint position, and not the drive targets (_currentDrivePos).